src/lib/fields.ts

```diff
@@ -0,0 +1,320 @@
+
+  await replaceField(record, fieldName, (value) => {
+    if (Array.isArray(value) || Array.isArray(newValue)) {
+      const list = [...(value || [])];
+
+      ((newValue ?? []) as any[]).forEach((e) => {
+        const existing = list.findIndex((item) => item.id == e?.id);
+        if (existing > -1) {
+          list.splice(existing, 1, e);
+        } else {
+          list.push(e);
+        }
+      });
+
+      return list;
+    } else if (typeof newValue === 'object') {
+      return {
+        ...value,
+        ...newValue
+      };
+    } else {
+      return newValue;
+    }
+  });
+};
+
+/**
+ * Append a field in record
+ *
+ * @param fieldName
+ * @param newValue
+ * @returns
+ */
+export const appendFieldToAccount = (
+  fieldName: IAccountExtensionField,
+  newValue: IAccountExtensionFields[IAccountExtensionField]
+) => {
+  return appendField(fieldName, newValue, aha.account);
+};
+
+/**
+ * Append a field in account
+ *
+ * @param record
+ * @param fieldName
+ * @param newValue
+ * @returns
+ */
+export const appendFieldToRecord = async (
+  record: Aha.RecordUnion,
+  fieldName: IRecordExtensionField,
+  newValue: IRecordExtensionFields[IRecordExtensionField]
+) => {
+  return appendField(fieldName, newValue, record);
+};
+
+/**
+ * Replace Extension Fields
+ *
+ * @param record
+ * @param fieldName
+ * @param replacer
+ */
+const replaceField = async (
+  record: Aha.RecordUnion | Aha.Account,
+  fieldName: IAccountExtensionField | IRecordExtensionField,
+  replacer: (val: any) => any
+) => {
+  const fieldValue = await record.getExtensionField(IDENTIFIER, fieldName);
+  const newValue = await replacer(fieldValue);
+  await record.setExtensionField(IDENTIFIER, fieldName, newValue);
+};
+
+/**
+ * Generate PR Id for account
+ *
+ * @param number
+ * @param ref
+ * @returns
+ */
+const accountPRId = (number?: string | number, ref?: string | number) => {
+  return [number, ref].join('_');
+};
+
+/**
+ *
+ * @param PR
+ * @returns
+ */
+export const AzureDevopsPRToPRLink = (pr: AzureDevops.PR): IExtensionFieldPullRequest => {
+  return {
+    id: pr.mergeId ?? '',
+    title: pr.title ?? '',
+    webUrl: `${pr?.repository?.webUrl}/pullrequest/${pr?.pullRequestId}`,
+    state: pr.status,
+    sourceBranch: pr?.sourceRefName,
+    targetBranch: pr?.targetRefName,
+    projectId: pr?.repository?.project?.id,
+    projectWebUrl: pr?.repository?.project?.url
+  };
+};
+
+/**
+ * Link Merge Request to Aha Record
+ *
+ * @param PR
+ * @returns
+ */
+export const linkMergeRequest = async (pr: AzureDevops.PR) => {
+  let record;
+  if (pr?.mergeId) {
+    record = await referenceToRecordFromId(pr.mergeId);
+  }
+
+  if (!record && pr?.sourceRefName) {
+    record = await referenceToRecordFromTitle(pr?.sourceRefName?.replace('refs/heads/', '') ?? '');
+  }
+
+  if (record) {
+    await linkPullRequestToRecord(pr, record);
+  }
+
+  return record;
+};
```

Comment:
@sorza-jimmy This appears to search the branch name only. Could you revise it to search by PR title as well?